private/buf/cmd/buf/command/registry/whoami/whoami.go

```diff
@@ -0,0 +1,122 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package whoami
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"connectrpc.com/connect"
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/buf/bufprint"
+	"github.com/bufbuild/buf/private/bufpkg/bufconnect"
+	"github.com/bufbuild/buf/private/gen/proto/connect/buf/alpha/registry/v1alpha1/registryv1alpha1connect"
+	registryv1alpha1 "github.com/bufbuild/buf/private/gen/proto/go/buf/alpha/registry/v1alpha1"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appext"
+	"github.com/bufbuild/buf/private/pkg/connectclient"
+	"github.com/bufbuild/buf/private/pkg/netext"
+	"github.com/spf13/pflag"
+)
+
+const (
+	formatFlagName = "format"
+)
+
+func NewCommand(
+	name string,
+	builder appext.SubCommandBuilder,
+) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name + " <domain>",
+		Short: `Check if you are logged in to the Buf Schema Registry`,
+		Long: `This command checks if you are currently logged into the Buf Schema Registry at the provided <domain>.
+The <domain> argument will default to buf.build if not specified.`,
+		Args: appcmd.MaximumNArgs(1),
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appext.Container) error {
+				return run(ctx, container, flags)
+			},
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+type flags struct {
+	Format string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(
+		&f.Format,
+		formatFlagName,
+		bufprint.FormatText.String(),
+		fmt.Sprintf(`The output format to use. Must be one of %s`, bufprint.AllFormatsString),
+	)
+}
+
+func run(
+	ctx context.Context,
+	container appext.Container,
+	flags *flags,
+) error {
+	remote := bufconnect.DefaultRemote
+	if container.NumArgs() == 1 {
+		remote = container.Arg(0)
+		if _, err := netext.ValidateHostname(remote); err != nil {
+			return err
+		}
+	}
+	clientConfig, err := bufcli.NewConnectClientConfig(container)
+	if err != nil {
+		return err
+	}
+	authnService := connectclient.Make(clientConfig, remote, registryv1alpha1connect.NewAuthnServiceClient)
+	currentUserResponse, err := authnService.GetCurrentUser(ctx, connect.NewRequest(&registryv1alpha1.GetCurrentUserRequest{}))
+	if err != nil {
+		if connectErr := new(connect.Error); errors.As(err, &connectErr) && connectErr.Code() == connect.CodeUnauthenticated {
+			return fmt.Errorf("Not currently logged in for %s", remote)
+		}
+		return err
+	}
+	user := currentUserResponse.Msg.User
+	if user == nil {
+		return errors.New("No user found for provided token")
```

Comment:
This mirrors the error that is returned from `buf registry login` when we check the token:
https://github.com/bufbuild/buf/blob/81698bbe0296d70ed886338aa943c28398c53f56/private/buf/cmd/buf/command/registry/registrylogin/registrylogin.go#L212-L214

We can prompt the user to get a new token using `buf registry login`. If that makes sense, then I can update the error message on both sides.

Reply:
I didn't provide a token though. Like as a user, this command does not accept a token, so it doesn't make sense to tell the user that their provided token didn't result in a user.

I get that this is the error message for the other one, but this just means both need to be updated, and should in no way reference tokens, as the user does not interact with tokens (typically) via either this command of `buf registry login`.

Reply:
That is fair -- I left the `buf registry login` error as-is, since it's part of a validation flow where the token is provided through the command. For this one, I provided instructions to the user to refresh their credentials and to check the env var if set.

Reply:
Still waiting on this

Reply:
For `buf registry login`, the error message comes up after the token is provided through the login process, so it should still make sense... in the case of `buf registry login`, it might make sense to report it as a syserror, since the token is provided by the user.